justspacesapp/urls.py

```diff
@@ -13,9 +13,98 @@
 
 urlpatterns = [
-    url(r'^$', CreateNewSurvey.as_view(), name='create-new-survey'),
+    url(r'^$',
+        SurveyListView.as_view(),
+        name='home'),
+
+    url(r'agencies/create$',
+        AgencyCreate.as_view(),
+        name='agencies-create'),
+
+    url(r'locations/create$',
+        LocationCreate.as_view(),
+        name='locations-create'),
+
+    url(r'studies/create$',
+        StudyCreate.as_view(),
+        name='studies-create'),
+
+    # Create new survey
+    url(r'surveys/create$',
+        view=fobi.views.create_form_entry,
+        name='fobi.create_form_entry'),
+
+    # Create survey element
+    url(_(r'^surveys/elements/create/(?P<form_entry_id>\d+)/'
+          r'(?P<form_element_plugin_uid>[\w_\-]+)/$'),
+        view=fobi.views.add_form_element_entry,
+        name='fobi.add_form_element_entry'),
+
+    # Edit survey element
+    url(_(r'^surveys/elements/edit/(?P<form_element_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_element_entry,
+        name='fobi.edit_form_element_entry'),
+
+    # Delete survey element
+    url(_(r'^surveys/elements/delete/(?P<form_element_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_element_entry,
+        name='fobi.delete_form_element_entry'),
+
+    # Create survey handler
+    url(_(r'^surveys/handlers/create/(?P<form_entry_id>\d+)/'
+          r'(?P<form_handler_plugin_uid>[\w_\-]+)/$'),
+        view=fobi.views.add_form_handler_entry,
+        name='fobi.add_form_handler_entry'),
+
+    # Edit survey handler
+    url(_(r'^surveys/handlers/edit/(?P<form_handler_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_handler_entry,
+        name='fobi.edit_form_handler_entry'),
+
+    # Delete survey handler
+    url(_(r'^surveys/handlers/delete/(?P<form_handler_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_handler_entry,
+        name='fobi.delete_form_handler_entry'),
+
+    # Edit survey
+    url(_(r'^surveys/edit/(?P<form_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_entry,
+        name='fobi.edit_form_entry'),
+
+    # Delete survey
+    url(_(r'^surveys/delete/(?P<form_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_entry,
+        name='delete_form_entry'),
+
+    # Survey list edit
+    url(r'surveys/edit$',
+        SurveyListEdit.as_view(),
+        name='survey-list-edit'),
+
+    # Survey list view
+    url(r'surveys/view$',
+        SurveyListView.as_view(),
+        name='survey-list-view'),
+
+    # Survey detail
+    url(_(r'^surveys/view/(?P<form_entry_slug>[\w_\-]+)/$'),
+        view=fobi.views.view_form_entry,
+        name='fobi.view_form_entry'),
+
+    # Submitted survey generic
+    url(_(r'^surveys/view/submitted/$'),
+        view=fobi.views.form_entry_submitted,
+        name='fobi.form_entry_submitted'),
+
+    # Submitted survey detail
+    url(_(r'^surveys/view/(?P<form_entry_slug>[\w_\-]+)/submitted/$'),
+        view=fobi.views.form_entry_submitted,
+        name='fobi.form_entry_submitted'),
```

Comment:
Overall, I think we have a good start on RESTful routes. I noted a couple places that require revision, but please plan to revise in another PR. 

For now, please add the requested doc string (cf. above).